src/run.c

```diff
@@ -538,28 +618,31 @@ Port io_dl_call(Net* net, Book* book, Port argm) {
   Port (*func)(Net*, Book*, Port) = dlsym(dl, symbol.buf);
   char* error = dlerror();
   if (error != NULL) {
-    fprintf(stderr, "io_dl_call: failed to get symbol '%s': %s\n", symbol.buf, error);
+    return inject_err_str(net, error);
   }
 
-  return func(net, book, tup.elem_buf[2]);
+  return inject_ok(net, func(net, book, tup.elem_buf[2]));
 }
 
 // Closes a loaded dylib, reclaiming the handle.
+//
+// Returns:  Result<*, String | *>
 Port io_dl_close(Net* net, Book* book, Port argm) {
   void* dl = readback_dylib(argm);
   if (dl == NULL) {
     fprintf(stderr, "io_dl_close: invalid handle\n");
-    return new_port(ERA, 0);
+
+    return inject_err(net, new_port(ERA, 0));
```

Comment:
Can you return a string instead of an Era in this case?
I feel like this should be a recoverable/debuggable error

Reply:
I'm going to return an `IOError/Type` meaning an invalid argument.